src/main/java/org/opensearch/performanceanalyzer/rca/store/rca/hot_node/ThreadMetricsRca.java

```diff
@@ -0,0 +1,385 @@
+    public static final double HIGH_BLOCKED_TIME_THRESHOLD_IN_SECONDS = 5d;
+    private final int rcaPeriod;
+    private final Thread_Blocked_Time threadBlockedTime;
+    private final Thread_Waited_Time threadWaitedTime;
+    private final Clock clock;
+    @VisibleForTesting final List<ThreadAnalysis> threadAnalyses;
+
+    public ThreadMetricsRca(
+            final Thread_Blocked_Time threadBlockedTime,
+            final Thread_Waited_Time threadWaitedTime,
+            final int rcaPeriodInSeconds) {
+        super(rcaPeriodInSeconds);
+        this.rcaPeriod = rcaPeriodInSeconds;
+        this.threadBlockedTime = threadBlockedTime;
+        this.threadWaitedTime = threadWaitedTime;
+        threadAnalyses = new ArrayList<>();
+        initThreadAnalyses();
+        this.clock = Clock.systemUTC();
+    }
+
+    private void initThreadAnalyses() {
+        // transport threads
+        threadAnalyses.add(
+                new ThreadAnalysis(
+                        s -> s.contains("transport"),
+                        BLOCKED_TRANSPORT_THREAD_COUNT,
+                        WAITED_TRANSPORT_THREAD_COUNT,
+                        MAX_TRANSPORT_THREAD_BLOCKED_TIME,
+                        MAX_TRANSPORT_THREAD_WAITED_TIME));
+    }
+
+    @Override
+    public void generateFlowUnitListFromWire(FlowUnitOperationArgWrapper args) {
+        final List<FlowUnitMessage> flowUnitMessages =
+                args.getWireHopper().readFromWire(args.getNode());
+        final List<ResourceFlowUnit<HotNodeSummary>> flowUnitList = new ArrayList<>();
+        LOG.debug("rca: Executing fromWire: {}", this.getClass().getSimpleName());
+        for (FlowUnitMessage flowUnitMessage : flowUnitMessages) {
+            flowUnitList.add(ResourceFlowUnit.buildFlowUnitFromWrapper(flowUnitMessage));
+        }
+        setFlowUnits(flowUnitList);
+    }
+
+    @Override
+    public void readRcaConf(RcaConf conf) {}
+
+    /**
+     * This RCA will be consumed by hot node rca as threads which are blocked beyond a threshold
+     * indicate a hot node. Until we calibrate a reasonable value for threshold of blocked time in
+     * threads we will set the RCA as healthy to ensure we don't hinder the decisioning of hot node
+     * rca.
+     */
+    @Override
+    public ResourceFlowUnit<HotNodeSummary> operate() {
+        try {
+            InstanceDetails instanceDetails = getInstanceDetails();
+            long currentTimeMillis = this.clock.millis();
+            LOG.debug("ThreadMetricsRca run at {}", currentTimeMillis);
+            collateThreadMetricData(currentTimeMillis);
+            publishStats();
+            return new ResourceFlowUnit<>(
+                    clock.millis(),
+                    new ResourceContext(Resources.State.HEALTHY),
+                    new HotNodeSummary(
+                            instanceDetails.getInstanceId(), instanceDetails.getInstanceIp()),
+                    false);
```

Comment:
Any reason why we dont want to persist this?

Reply:
We are return healthy in every case now and hence there won't be much value in persisting now. Once we calibrate thresholds for blocked time and count of threads exceeding threshold to signify hot node scenario we can persist this.